src/Server/HTTPHandler.cpp

```diff
@@ -968,6 +981,14 @@ HTTPRequestHandlerFactoryPtr createPredefinedHandlerFactory(IServer & server,
     Poco::Util::AbstractConfiguration::Keys headers_name;
     config.keys(config_prefix + ".headers", headers_name);
 
+    std::optional<BasicCredentials> config_credentials;
+    if (config.has(config_prefix + ".handler.user") || config.has(config_prefix + ".handler.password"))
```

Comment:
If `handler.user` isn't in the config then it doesn't seem right to still get password from the config.

Reply:
Yes, but, I did this because even though specifying only `password` is a bad idea, ignoring it completely is worse